src/components/Nav/NewCollectiveModal.vue

```diff
@@ -0,0 +1,363 @@
+						class="modal-buttons-right"
+						@click="advanceToMembers">
+						{{ t('collectives', 'Add people') }}
+					</NcButton>
+				</div>
+			</div>
+
+			<div v-else-if="state === 1" class="modal-collective-wrapper">
+				<h2 class="modal-collective-title">
+					{{ t('collectives', 'Members') }}
+				</h2>
+
+				<div class="modal-collective-members">
+					<MemberPicker :selection-set="selectedMembers"
+						@updateSelection="updateSelectedMembers" />
+				</div>
+
+				<div class="modal-buttons">
+					<NcButton @click="state = 0">
+						{{ t('collectives', 'Back') }}
+					</NcButton>
+					<NcButton type="primary"
+						:disabled="loading"
+						class="modal-buttons-right"
+						@click="onCreate">
+						{{ t('collectives', 'Create') }}
+					</NcButton>
+				</div>
+			</div>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import { mapActions, mapGetters, mapState } from 'vuex'
+import { ADD_MEMBERS_TO_CIRCLE, GET_CIRCLES, NEW_COLLECTIVE } from '../../store/actions.js'
+import displayError from '../../util/displayError.js'
+import { NcButton, NcEmojiPicker, NcEmptyContent, NcModal, NcSelect, NcTextField } from '@nextcloud/vue'
+import AlertCircleOutlineIcon from 'vue-material-design-icons/AlertCircleOutline.vue'
+import CirclesIcon from '../Icon/CirclesIcon.vue'
+import CloseIcon from 'vue-material-design-icons/Close.vue'
+import CollectivesIcon from '../Icon/CollectivesIcon.vue'
+import MemberPicker from '../Member/MemberPicker.vue'
+import { showError } from '@nextcloud/dialogs'
+
+export default {
+	name: 'NewCollectiveModal',
+
+	components: {
+		AlertCircleOutlineIcon,
+		CirclesIcon,
+		CloseIcon,
+		CollectivesIcon,
+		MemberPicker,
+		NcButton,
+		NcEmojiPicker,
+		NcEmptyContent,
+		NcModal,
+		NcSelect,
+		NcTextField,
+	},
+
+	data() {
+		return {
+			circle: null,
+			emoji: null,
+			loading: false,
+			name: '',
+			nameExists: '',
+			pickCircle: false,
+			selectedMembers: {},
+			state: 0,
+		}
+	},
+
+	computed: {
+		...mapState({
+			updatedCollective: (state) => state.collectives.updatedCollective,
+		}),
+
+		...mapGetters([
+			'availableCircles',
+			'collectiveChanged',
+			'randomCollectiveEmoji',
+			'updatedCollectivePath',
+		]),
+
+		circles() {
+			return this.availableCircles.map(c => c.sanitizedName)
+		},
+
+		anyCircle() {
+			return this.circles.length > 0
+		},
+
+		newCollectiveName() {
+			return this.pickCircle ? this.circle : this.name
+		},
+
+		nameIsTooShort() {
+			return !!this.newCollectiveName && this.newCollectiveName.length < 3
+		},
+
+		nameIsTaken() {
+			return !!this.newCollectiveName && this.nameExists === this.newCollectiveName
+		},
+
+		nameIsInvalid() {
+			return this.nameIsTooShort || this.nameIsTaken
+		},
+
+		nameError() {
+			if (this.nameIsTooShort) {
+				return t('collectives', 'Name too short, requires at least three characters')
+			} else if (this.nameIsTaken) {
+				return t('collectives', 'A collective with this name already exists')
+			}
+			return null
```

Comment:
If you don't return anything `undefined` will be returned - which should be just as good as null here.

```suggestion
```